ode-essentials/ode/filesystem/MemoryFileSystem.h

```diff
@@ -0,0 +1,64 @@
+
+#pragma once
+
+#include <vector>
+#include <string>
+#include <optional>
+
+#include "FilePath.h"
+#include "../utils.h"
+
+namespace ode {
+
+/// Representation of a filesystem loaded to memory.
+class MemoryFileSystem {
+public:
+    enum class Error : uint8_t {
```

Comment:
It isn't critically important. I tend to make enum classes of a smaller size to signal the number of elements in them is and will remain limited.

Reply:
This makes sense when having very large arrays with this type to save memory but otherwise using uint8_t is actually less efficient. Also I don't think I've ever seen an enum with more than 256 values. ;)